### PR TITLE
CORDA-3171  Hello world tutorial docs code fixed for Corda v4

### PR DESCRIPTION
**Pull request by barrylapthorn**
Added the `BelongsToContract` annotation to code in docs which is required for
`ContractState` subclasses to associate them with a Contract subclass.

Fixed various typos.

### CODE REVIEW

File: docs/source/hello-world-running.rst [hunk 1]
```diff
@@ -153,7 +153,7 @@ The vaults of PartyA and PartyB should both display the following output:
     - ref:
         txhash: "5CED068E790A347B0DD1C6BB5B2B463406807F95E080037208627565E6A2103B"
         index: 0
-      contractStateClassName: "com.template.state.IOUState"
+      contractStateClassName: "com.template.states.IOUState"
```
Comment (zkiss) [inline docs/source/hello-world-running.rst]: Shouldn't the package be `net.corda.docs.java.tutorial.helloworld`?

File: docs/source/hello-world-state.rst [hunk 2]
```diff
@@ -77,6 +77,7 @@ If you're following along in Java, you'll also need to rename ``TemplateState.ja
 
 To define ``IOUState``, we've made the following changes:
 
+* We've added the annotation ``@BelongsToContract`` as a contract state must be tied to a contract since Corda v4
```
Comment (lankydan) [inline docs/source/hello-world-state.rst:80]: remove the "since Corda v4"

File: docs/source/hello-world-running.rst (hunk 1 above)
Comment (lankydan) [reply docs/source/hello-world-running.rst:156]: this is true

File: docs/source/hello-world-flow.rst [hunk 3]
```diff
@@ -80,7 +80,7 @@ annotation out will lead to some very weird error messages!
 
 There are also a few more annotations, on the ``FlowLogic`` subclass itself:
 
-  * ``@InitiatingFlow`` means that this flow is part of a flow pair and that it triggers the other side to run the
+  * ``@InitiatingFlow`` means that this flow is part of a flow pair and that it triggers the other side to run
     the counterpart flow (which in our case is the ``IOUFlowResponder`` defined below).
```
Comment (lankydan) [inline docs/source/hello-world-flow.rst:84]: "the counterpart flow" -> "the counterparty flow

**Review by lankydan — COMMENTED** [review]
Sorry about not reviewing this for so long... kept forgetting...

**Comment by barrylapthorn** [comment]
Not merging at this time.

**Pull request closed without merging**